fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalHudiScan.java

```diff
@@ -62,8 +62,7 @@ public class LogicalHudiScan extends LogicalFileScan {
     private static final Logger LOG = LogManager.getLogger(LogicalHudiScan.class);
 
     // for hudi incremental read
-    private final Optional<TableScanParams> scanParams;
-    private final Optional<IncrementalRelation> incrementalRelation;
+    private Optional<IncrementalRelation> incrementalRelation;
```

Comment:
In Nereids, most of all objects are required to be immutable.
So here we should retain the `final` qualifier. And create a new one if we need to modify a field in class